File: client/src/types/package.ts
export enum PackageStatus {
  Created = 0,
  Sent = 1,
  Accepted = 2,
  Returned = 3,
  Canceled = 4,
}

export const PackageStatusLabel: Record<PackageStatus, string> = {
  [PackageStatus.Created]: "Created",
  [PackageStatus.Sent]: "Sent",
  [PackageStatus.Accepted]: "Accepted",
  [PackageStatus.Returned]: "Returned",
  [PackageStatus.Canceled]: "Canceled",
};

export interface StatusHistory {
  id: string;
  packageId: string;
  status: PackageStatus;
  changedAt: string;
  description?: string;
}

export interface PackageListItem {
  id: string;
  trackingNumber: string;
  senderName: string;
  recipientName: string;
  status: PackageStatus;
  createdAt: string;
}

export interface StatusItem {
  status: PackageStatus;
  timestamp: string;
}

export interface PackageDetails extends PackageListItem {
  status: PackageStatus;
  senderAddress: string;
  senderPhone: string;
  recipientAddress: string;
  recipientPhone: string;
  statusHistory: StatusHistory[];
  createdAt: string;
  allowedTransitions: PackageStatus[];
}

export interface CreatePackageInput {
  senderName: string;
  senderAddress: string;
  senderPhone: string;
  recipientName: string;
  recipientAddress: string;
  recipientPhone: string;
  trackingNumber: string;
}

File: server/Models/Package.cs
using System.ComponentModel.DataAnnotations;
using Server.Models.Enums;

namespace Server.Models
{
    public class Package
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        [MaxLength(20)]
        public string TrackingNumber { get; set; } = default!;

        // Sender Info
        [Required, MaxLength(100)]
        public string SenderName { get; set; } = default!;

        [Required, MaxLength(200)]
        public string SenderAddress { get; set; } = default!;

        [Required, Phone, MaxLength(20)]
        public string SenderPhone { get; set; } = default!;

        // Recipient Info
        [Required, MaxLength(100)]
        public string RecipientName { get; set; } = default!;

        [Required, MaxLength(200)]
        public string RecipientAddress { get; set; } = default!;

        [Required, Phone, MaxLength(20)]
        public string RecipientPhone { get; set; } = default!;

        public PackageStatus Status { get; set; } = PackageStatus.Created;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation property
        public ICollection<StatusHistory> History { get; set; } = new List<StatusHistory>();
    }
}
